third_party/2/pycurl.pyi

```diff
@@ -39,20 +40,26 @@ class CurlMulti(object):
     def perform(self) -> Tuple[Any, int]: ...
     def fdset(self) -> tuple: ...
     def select(self, timeout: float = ...) -> int: ...
-    def info_read(self, max_objects: int) -> tuple: ...
+    def info_read(self, max_objects: Optional[int]) -> tuple: ...
```

Comment:
I think you want `max_objects: int = ...` instead. What you wrote means that the `max_objects` argument must be given, but it can be either an int or None; but in reality info_read takes either 0 or 1 arguments, and the one argument must be an int.

Reply:
I see, thanks for clarifying.

I wasn't able to find docs on this. Could you point me to them? If not, is there somewhere I can open a PR to help document the difference?

Reply:
There are three main pieces of documentation: PEP 484 (https://www.python.org/dev/peps/pep-0484/), CPython's typing documentation (https://docs.python.org/3/library/typing.html), and mypy's docs (http://mypy.readthedocs.io/en/stable/).

CPython's doc actually address this difference: https://docs.python.org/3/library/typing.html#typing.Optional. Maybe it also deserves a mention in mypy.

Reply:
Got it. This part helps clear up the difference for me https://www.python.org/dev/peps/pep-0484/#arbitrary-argument-lists-and-default-argument-values